cloud-sql/mysql/Startup.cs

```diff
@@ -45,48 +43,105 @@ public Startup(IConfiguration configuration)
-            {
-                services.AddGoogleExceptionLogging(options =>
-                {
-                    options.ProjectId = projectId;
-                    options.ServiceName = "CloudSqlSample";
-                    options.Version = "Test";
-                });
-            }
-            services.AddScoped(typeof(DbConnection), (IServiceProvider) =>
-                InitializeDatabase());
+            services.AddSingleton(sp => new MySqlConnection().GetMySqlConnectionString());
             services.AddMvc(options =>
             {
                 options.Filters.Add(typeof(DbExceptionFilterAttribute));
             });
-            _services = services;
         }
 
-        DbConnection InitializeDatabase()
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env,
+            ILoggerFactory loggerFactory)
         {
-            DbConnection connection;
+            loggerFactory.AddConsole(Configuration.GetSection("Logging"));
+
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+            else
+            {
+                // Configure error reporting service.
+                app.UseGoogleExceptionLogging();
+                app.UseExceptionHandler("/Home/Error");
+            }
+
+            app.UseMvc(routes =>
+            {
+                routes.MapRoute(
+                    name: "default",
+                    template: "{controller=Home}/{action=Index}/{id?}");
+            });
+        }
+    }
+
+    static class StartupExtensions
+    {
+        public static DbConnection OpenWithRetry(this DbConnection connection) =>
+            // [START cloud_sql_mysql_dotnet_ado_backoff]
+            Policy
+                .HandleResult<DbConnection>(conn => conn.State != ConnectionState.Open)
+                .WaitAndRetry(new[]
+                {
+                    TimeSpan.FromSeconds(1),
+                    TimeSpan.FromSeconds(2),
+                    TimeSpan.FromSeconds(5)
+                }, (result, timeSpan, retryCount, context) =>
+                {
+                    // Log any warnings here.
+                    if(result.Exception != null && retryCount == 3)
+                    {
+                        throw result.Exception;
+                    }
+                })
+                .Execute(() =>
+                {
+                    // Return a new connection.
+                    try {
+                        connection.Open();
+                    }
+                    catch (MySqlException e)
+                    {
+                         Console.WriteLine(
+                             $"Error connecting to database: {e.Message}");
+                    }
+                    return connection;
+                });
+            // [END cloud_sql_mysql_dotnet_ado_backoff]

```

Comment:
And now what we are roughly doing is the same as Polly would do like this:
```suggestion
        public static void OpenWithRetry(this DbConnection connection) =>
            // [START cloud_sql_mysql_dotnet_ado_backoff]
            Policy
                .Handle<MySqlException>()
                .WaitAndRetry(new[]
                {
                    TimeSpan.FromSeconds(1),
                    TimeSpan.FromSeconds(2),
                    TimeSpan.FromSeconds(5)
                })
                .Execute(() => connection.Open());
            // [END cloud_sql_mysql_dotnet_ado_backoff]

```

Reply:
Done.